db/index.go

```diff
@@ -88,7 +88,7 @@ func indexUpdate(baseKey ds.Key, path string, index Index, tx ds.Txn, key ds.Key
 		return nil
```

Comment:
Indexes where relying on the _prefix query will contain non-child keys_ assumption.